app/models/mixins/supports_feature_mixin.rb

```diff
@@ -0,0 +1,56 @@
+module SupportsFeatureMixin
+  extend ActiveSupport::Concern
+
+  class_methods do
+    def supports(feature, is_supported = true, reason = "", &block)
+      method_name = "supports_#{feature}?"
+
+      self.class.send(:define_method, method_name) do
+        result = if is_supported
+                   true
+                 else
+                   reason
+                 end
+        if result === true
+          true
+        else
+          false
+        end
```

Comment:
(For fun, but I still think we should change the API to return something more informative)

```
self.class.send(:define_method, method_name) do
  result = if is_supported
             true
           else
             reason
           end
  if result === true
    true
  else
    false
  end
end

# =>

self.class.send(:define_method, method_name) do
  result = is_supported || reason   # Changed
  if result === true
    true
  else
    false
  end
end

# =>

self.class.send(:define_method, method_name) do
  result = is_supported || reason
  result === true  # Changed
end

# =>

self.class.send(:define_method, method_name) do
  is_supported   # Changed. Reason removed because it’s never === true
end
```